### DISCUSSION TITLE
GithubAPI functioning

### DISCUSSION THREAD

**Discussion by Naoufal69** [community/community#61608] [Apps, API and Webhooks]
### Select Topic Area

Bug

### Body

Hello,
First, i want to say, i'm sorry for my poor english. I'm not a native speaker.

So, here is my problem. It seem that i can't reach my repo through github api. It's a private repo. Here is my token rights : 

![img2](https://github.com/community/community/assets/94354202/d4fc979a-fe27-44a3-8aaa-48af3c6dcaa9)
![img1](https://github.com/community/community/assets/94354202/11c767da-9276-4550-9ea2-3ed919d6a9ad)

Here is my code that return a 404
<code>
require("dotenv").config();
const { Octokit } = require("octokit");
const fetch = require("node-fetch");

const GITHUB_USERNAME = process.env.GITHUB_USERNAME;
const GITHUB_REPO = "git@github.com:Naoufal69/API-Rhone.git";
const GITHUB_BRANCH = "main";
const GITHUB_TOKEN = process.env.GITHUB_TOKEN; // ou votre mot de passe si vous ne voulez pas utiliser de token

const octokit = new Octokit({
  request: {
    fetch,
  },
  auth: GITHUB_TOKEN,
});

const getCommits = async () => {
  try {
    const response = await octokit.request(
      "GET /repos/{owner}/{repo}/commits",
      {
        owner: GITHUB_USERNAME,
        repo: GITHUB_REPO,
        headers: {
          "X-GitHub-Api-Version": "2022-11-28",
        },
      }
    );
    console.log("Response : " + response.data);
  } catch (error) {
    console.log(error);
  }
};

getCommits();

</code>

As I'm a begginer, both in github api and in this forum, i'm sorry if i didn't explain my problem well enough, or if it was the wrong way to do it.

PS : I'm sure that i have the good data in my `.env` file

**Comment by nirav-gajera** [comment]
In the` GITHUB_REPO` constant, you are using the SSH URL of the repository (`git@github.com:Naoufal69/API-Rhone.git`). The Octokit API expects the repository name in the format` owner/repo`, and using an SSH URL will not work with the API.

-modify your code to use the HTTPS URL format:
```
require("dotenv").config();
const { Octokit } = require("@octokit/rest");
const fetch = require("node-fetch");

const GITHUB_USERNAME = process.env.GITHUB_USERNAME;
const GITHUB_REPO = "Naoufal69/API-Rhone"; // Provide the repository name in owner/repo format
const GITHUB_BRANCH = "main";
const GITHUB_TOKEN = process.env.GITHUB_TOKEN;

const octokit = new Octokit({
  auth: GITHUB_TOKEN,
});

const getCommits = async () => {
  try {
    const response = await octokit.request("GET /repos/{owner}/{repo}/commits", {
      owner: GITHUB_USERNAME,
      repo: GITHUB_REPO,
      headers: {
        "X-GitHub-Api-Version": "2022-11-28",
      },
    });
    console.log("Response:", response.data);
  } catch (error) {
    console.log(error);
  }
};

getCommits();

```

**Comment by Naoufal69** [comment]
> In the` GITHUB_REPO` constant, you are using the SSH URL of the repository (`git@github.com:Naoufal69/API-Rhone.git`). The Octokit API expects the repository name in the format` owner/repo`, and using an SSH URL will not work with the API.

Thank you for your input. I made the change, but still, i'm getting 404

**Reply by nirav-gajera** [reply]
Retry with REST API Endpoint: Instead of using the shorthand method `octokit.request,` try using the full REST API endpoint format:

```
const response = await octokit.rest.repos.listCommits({
  owner: GITHUB_USERNAME,
  repo: GITHUB_REPO,
  headers: {
    "X-GitHub-Api-Version": "2022-11-28",
  },
});

```

**Reply by Naoufal69** [reply]
Still 404, what could have changed with the full endpoint format ?

**Reply by nirav-gajera** [reply]
```
const response = await octokit.request("GET /repos/{owner}/{repo}/commits", {
      owner: GITHUB_USERNAME,
      repo: GITHUB_REPO,
      headers: {
        "X-GitHub-Api-Version": "2022-11-28",
      },
    });
```

to

```
const response = await octokit.rest.repos.listCommits({
  owner: GITHUB_USERNAME,
  repo: GITHUB_REPO,
  headers: {
    "X-GitHub-Api-Version": "2022-11-28",
  },
});
```

-and make sure you replaced owner and repo name  instead of GITHUB_USERNAME and  GITHUB_REPO

**Reply by Naoufal69** [reply]
What do you mean by replaced ?
I have did what you told me to do, but it's not working, is it because of my token ?

**Reply by nirav-gajera** [reply]
yes, might be related to your Personal Access Token (PAT) not having the necessary permissions to access the private repository. The repo scope is required to access private repositories through the GitHub API.

try with new token

**Reply by Naoufal69** [reply]
I have tried to create a new token, nothing, i'm loosing hope

**Comment by Naoufal69** [comment]
Any others suggest ?
